wandb/sdk/launch/utils.py

```diff
@@ -794,3 +794,32 @@ def recursive_macro_sub(source: Any, sub_dict: Dict[str, Optional[str]]) -> Any:
         }
     else:
         return source
+
+
+def fetch_and_validate_template_variables(
+    runqueue: Any, fields: dict
+) -> Dict[str, Any]:
+    template_variables = {}
+
+    variable_schemas = {}
+    for tv in runqueue.template_variables:
+        variable_schemas[tv["name"]] = json.loads(tv["schema"])
+
+    for field in fields:
+        key, val = field.split("=")
+        if key not in variable_schemas:
+            raise LaunchError(
+                f"Queue {runqueue.name} does not support overriding {key}."
+            )
+        schema = variable_schemas.get(key, {})
+        field_type = schema.get("type")
+        try:
+            if field_type == "integer":
+                val = int(val)
+            elif field_type == "number":
+                val = float(val)
```

Comment:
Not strictly necessary, but since we already fetching the schema and doing some validation we might as well just validate all of the constraints of the schema before we try to submit a job.

Reply:
I worry about having these same checks in two different places. It means any future change/addition to template var validation needs to be implemented twice. I'd rather keep the checks here as minimal as possible.